app/assets/javascripts/Components/markdown_edit.jsx

```diff
@@ -0,0 +1,78 @@
+import React from "react";
+import MarkdownPreview from "./markdown_preview";
+import {Tab, Tabs, TabList, TabPanel} from "react-tabs";
+
+import PropTypes from "prop-types";
+
+export default class MarkdownEdit extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      content: this.props.content,
+      annotation_text_id: "",
+    };
+  }
+
+  handleChange = event => {
+    this.setState({content: event.target.value});
+    this.props.handleChange(event);
+  };
+
+  render() {
+    return (
+      <Tabs disableUpDownKeys>
+        <TabList>
+          <Tab id="write-tab">{I18n.t("write")}</Tab>
+          <Tab id="preview-tab">{I18n.t("preview")}</Tab>
+        </TabList>
+        <TabPanel forceRender>
+          <label>
+            <textarea
+              required={true}
+              id="new_annotation_content"
```

Comment:
Part of making this component more reusable is removing reference to specific applications (e.g. "annotations"). For now you can make a prop that allows an id to be set, since it does look like the `new_annotation_content` id is necessary for the annotation modal autocomplete, at least for now.